internal/controller/bucket/bucket.go

```diff
@@ -150,105 +144,142 @@ func (c *connector) Connect(ctx context.Context, mg resource.Managed) (managed.E
 	}
 
 	return &external{
-			kubeClient:   c.kube,
-			backendStore: c.backendStore,
-			log:          c.log},
+			kubeClient:      c.kube,
+			autoPauseBucket: c.autoPauseBucket,
+			backendStore:    c.backendStore,
+			log:             c.log},
 		nil
 }
 
 // An ExternalClient observes, then either creates, updates, or deletes an
 // external resource to ensure it reflects the managed resource's desired state.
 type external struct {
-	kubeClient   client.Client
-	backendStore *backendstore.BackendStore
-	log          logging.Logger
+	kubeClient      client.Client
+	autoPauseBucket bool
+	backendStore    *backendstore.BackendStore
+	log             logging.Logger
 }
 
+//nolint:gocognit,gocyclo,cyclop // Function requires numerous checks.
 func (c *external) Observe(ctx context.Context, mg resource.Managed) (managed.ExternalObservation, error) {
 	bucket, ok := mg.(*v1alpha1.Bucket)
 	if !ok {
 		return managed.ExternalObservation{}, errors.New(errNotBucket)
 	}
 
-	if !c.backendStore.BackendsAreStored() {
-		return managed.ExternalObservation{}, errors.New(errNoS3BackendsStored)
+	if len(bucket.Status.AtProvider.BackendStatuses) == 0 {
+		return managed.ExternalObservation{
+			ResourceExists:   false,
+			ResourceUpToDate: true,
+		}, nil
+	}
+
+	available := false
+	for _, c := range bucket.Status.Conditions {
+		if c.Type == xpv1.TypeReady && c.Reason == xpv1.ReasonAvailable && c.Status == corev1.ConditionTrue {
+			available = true
+
+			break
+		}
+	}
+	if !available && !utils.IsHealthCheckBucket(bucket) {
+		return managed.ExternalObservation{
+			ResourceExists:   true,
+			ResourceUpToDate: false,
+		}, nil
 	}
 
-	type bucketExistsResult struct {
-		bucketExists bool
-		err          error
+	if len(bucket.Spec.Providers) == 0 {
+		bucket.Spec.Providers = c.backendStore.GetAllActiveBackendNames()
+	}
+
+	allBackendClients := c.backendStore.GetBackendClients(bucket.Spec.Providers)
+
+	missing := len(bucket.Spec.Providers)
+	for _, provider := range bucket.Spec.Providers {
+		if _, ok := allBackendClients[provider]; !ok {
+			missing--
```

Comment:
What's the reasoning behind this? If we're missing a client to the backend then we can't tell if it's up to date.

Reply:
Only deleted providers are inactive. It means deleted backends are not counted in the missing backends.